src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs

```diff
@@ -317,6 +318,12 @@ internal static void ApplyParameterInfo(this JsonNode schema, ApiParameterDescri
         {
             schema.ApplyRouteConstraints(constraints);
         }
+        // Parameters sourced from the header, query, and route parameter cannot be nullable.
+        if (parameterDescription.Source is { } bindingSource &&
+            (bindingSource == BindingSource.Header || bindingSource == BindingSource.Query || bindingSource == BindingSource.Path))
```

Comment:
Should form params also be excluded from being nullable?

Reply:
Looked into this. It appears that we'll want arguments bound from a form to also be excluded. I'll update the implementation and add tests.